public/content/posts/performance-improvement-in-a-next.js-page/index.md

```diff
@@ -0,0 +1,109 @@
+---
+title: "Performance improvement in a Next.js page"
+date: "2022-09-12T09:30:08.625Z"
+subtitle: "When improving the performance, web vitals take a vital place. In this article we are mainly looking at the LCP socre."
+imageSrc: performance.png
+imageAlt: Performance score in lighthouse
+author: gayan-witharana
+tags:
+  - performance
+  - web-development
+  - Next.js
+  - web-vitals
+---
+
+## Are web vitals necessary?
+
+From Google monthly user experience data, between May and October 2021, overall CWV compliance increased by 58 per cent worldwide.
+
+- Largest Contentful Paint – 44 per cent compliance
+- Cumulative Layout Shift – 71 per cent met requirements
+- First Input Delay – 91 per cent compliance
+
+So obviously if you're not optimizing for core web vitals, you are definitely in the minority.
+
+## LCP
+
+In the fishbrain web application, we're doing fine with the FID and Cls scores as you can see in the title image. But the LCP score is quite hight compared to the standard time (2.5s)
+
+![LCP score margin](/lcp.png "LCP Score")
+
+### Identify LCP of a page
+
+There are few ways we can identify the LCP of a page. One of easy was is to use latest Performance Insights tab in google chrome. Once you clicked "Measure page load" button it will run for a few seconds and then it will show some details related to performance. On the right side of the tab, you will be able to see web vitals and once you hover LCP it will highlight the LCP content of the page.
+
+![Google Chrome Performance Insights](/perf-insight.png "Google Chrome Performance Insights")
+
+### improve performance of the image
+
+Once we identify the LCP element of the page, we can try to make sure it loads fast relatively. Since in this case it's an image we can should load the image in "eager" mode instead of the "lazy" mode.
+
+```html
+<img src="imageUrl" loading="eager" />
+```
+
+since the eager is the default value for loading prop, in this case we can ignore the loading prop as well.
```

Comment:
```suggestion
Since `eager` is the default value for the loading attribute, in this case we can remove the loading prop as well.
```

I think attribute is clearer than prop in this instance. Although technically it is a prop, it's really just a layer over the HTML implementation, where the name attribute would be used instead.